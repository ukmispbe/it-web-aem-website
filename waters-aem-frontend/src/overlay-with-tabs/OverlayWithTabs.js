import React, { useState } from 'react';
import PropTypes from "prop-types";
import Modal, { Header, keys } from "../utils/modal";
import Tabs from "../navigation/tabs";
import './overlaywith.css';

const OverlayWithTabs = props => {
    const [state, setState] = useState({
        open: props.isOpen,
        gallerySelected: "",
        activeTabFilter: "",
        activeTabIndex: 0,
    });

    const handleClose = (e) => setState({ open: false });
    
    const handleGallerySelected = (e) => {
       
        let tabId;
        let activeTabFilterStatus = props.tabs[0].name;
        e.value || e.value === 0 ? (tabId = e.value) : (tabId = e);
    
        if (tabId === 1) {
          activeTabFilterStatus = props.tabs[1].name;
        }

        if(props.activeTabHandler) {
            props.activeTabHandler(tabId);
        }
    
        setState({
          activeTabFilter: activeTabFilterStatus,
          activeTabIndex: tabId,
          open: true
        });
      };

    const renderTabs = () => {
        const { tabs = [] } = {
          tabs: props.tabs,
        };
        return (
          <Tabs
            className="cmp-search__categories-tabs"
            items={tabs}
            activeIndex={state.activeTabIndex}
            onClick={(e) => handleGallerySelected(e)}
            enableFading={true}
          />
        );
      };

    return (
        <>
          <Modal
                isOpen={state.open}
                onClose={handleClose}
                className="cmp-asset-display-modal"
              >
              <div>
                  <Header title="" />
                  <div className="cmp-gallery-tab">{renderTabs()}</div>
                  <div>{props.children}</div>
              </div>                
            </Modal>
        </>
    )
}

OverlayWithTabs.propTypes = {
  isOpen: PropTypes.bool,
  tabs: PropTypes.arrayOf(PropTypes.object),
  activeTabHandler: PropTypes.func
}

OverlayWithTabs.defaultProps = {
  isOpen: false,
  tabs: [],
  activeTabHandler: () => {}
}

export default OverlayWithTabs;