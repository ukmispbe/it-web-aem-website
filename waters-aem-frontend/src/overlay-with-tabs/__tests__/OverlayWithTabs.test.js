import React from "react";
import { shallow } from "enzyme";
import { OverlayWithTabs } from "../OverlayWithTabs";
const defaultConfig = {
  tabs:[{ name: "Product Images" }, { name: "Video" }, { name: "Test" }], 
  children: '<div>data</div>', 
  isOpen : true
};

describe("<Thumbnail />", () => {
  let OverlayWithTabsComponent = "";
  beforeEach(() => {
    OverlayWithTabsComponent = shallow(<OverlayWithTabs {...defaultConfig} />);
  });

  test("should render correctly", () => {
    expect(OverlayWithTabsComponent).toMatchSnapshot();
  });

  test("should render image without label", () => {
    const props = {
      tabs:[], 
      children: '', 
      isOpen : false
    };
    const wrapper = shallow(<OverlayWithTabs {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
