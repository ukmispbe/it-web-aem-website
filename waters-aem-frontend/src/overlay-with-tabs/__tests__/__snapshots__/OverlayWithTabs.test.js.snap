// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OverlayWithTabs /> should render component without tabs 1`] = `ShallowWrapper {}`;

exports[`<OverlayWithTabs /> should render correctly 1`] = `ShallowWrapper {}`;
